registry/storage/driver/s3-aws/s3_test.go

```diff
@@ -48,12 +49,7 @@ func init() {
 		logLevel         = os.Getenv("S3_LOGLEVEL")
 	)
 
-	root, err := os.MkdirTemp("", "driver-")
-	if err != nil {
-		panic(err)
-	}
-	defer os.Remove(root)
```

Comment:
Wait, what? That was hilariously broken: it created a temporary directory, _then deleted it once `init()` returned!_ Your fixing the test suite to use a root directory _that exists_ might have something to do with why the s3 tests are now failing. It also suggests that the test failures might be legitimate bugs in the driver. Fun!

Reply:
Also, that `defer` that's supposed to be cleaning up the `root` dir would never get called if the create `panic`ked 🫠

Reply:
Of all the issues with the old code, that isn't one of them. There wouldn't be a directory in need of cleanup if `os.MkdirTemp` returned an error.

Reply:
Fair comment. Still, seeing it leaves a bad taste in my mouth as it passively encourages bad practice 🤷‍♂️

Reply:
What's the bad practice? Panic and `init()` notwithstanding, that is exactly the sort of code I would have written (inside a test function) to create a temporary directory in the days before `t.TempDir()` and `t.Cleanup()`.

The only bad practice would be a fallible, side-effectful function that is leaks side effects on failure. It would be the function, not the caller, that's bad. Consider:
```go
cleanup, err := DoEffectfulThing(...)
if err != nil {
    // caller should never be required to call cleanup() when err != nil.
    // The function might even return cleanup == nil when err != nil like
    // any other function which uses multi-returns as a pseudo-sum type.
    return err
}
defer cleanup()
```